orderer/consensus/etcdraft/node.go

```diff
@@ -201,6 +202,14 @@ func (n *node) send(msgs []raftpb.Message) {
 
 		status := raft.SnapshotFinish
 
+		// Replace node list in snapshot with CURRENT node list in cluster.
+		if msg.Type == raftpb.MsgSnap {
+			state := n.confState.Load()
+			if state != nil {
+				msg.Snapshot.Metadata.ConfState.Voters = state.(*raftpb.ConfState).Voters
```

Comment:
Why did you decide to only override the `Voters` and not the entire conf state?
I see that conf state has more fields, like voters outgoing:

```
message ConfState {
	// The voters in the incoming config. (If the configuration is not joint,
	// then the outgoing config is empty).
	repeated uint64 voters = 1;
	// The learners in the incoming config.
	repeated uint64 learners          = 2;
	// The voters in the outgoing config.
	repeated uint64 voters_outgoing   = 3;
	// The nodes that will become learners when the outgoing config is removed.
	// These nodes are necessarily currently in nodes_joint (or they would have
	// been added to the incoming config right away).
	repeated uint64 learners_next     = 4;
	// If set, the config is joint and Raft will automatically transition into
	// the final config (i.e. remove the outgoing config) when this is safe.
	optional bool   auto_leave        = 5 [(gogoproto.nullable) = false];
}
```

Etcd overrides the conf state fully, not just the voters.

Reply:
I took this patch from the work by @guoger on https://github.com/hyperledger/fabric/pull/2707/files#diff-b7eaca6d9a2722c748ff597e17957ee6eafac930654420eb9a82d7df7d7bbf0bR209

I can make it take the whole ConfState instead.

Reply:
Let's do it, it can't hurt :)

Reply:
ok, done!